base/PerfOptions.php

```diff
@@ -134,6 +180,50 @@ public function __construct(Vector<string> $argv) {
     $this->siege = hphp_array_idx($o, 'siege', 'siege');
     $this->nginx = hphp_array_idx($o, 'nginx', 'nginx');
 
+    $isFacebook = array_key_exists('fbcode', $o);
+    $fbcode = "";
+    if ($isFacebook) {
+      $val = hphp_array_idx($o, 'fbcode', false);
+      if (is_string($val) && $val !== '') {
+        $fbcode = $val;
+      } else {
+        $fbcode = getenv('HOME') . '/fbcode';
+      }
+    }
+    $this->waitAtEnd = array_key_exists('wait-at-end', $o);
+
+    $this->precompile  = array_key_exists('repo-auth', $o);
+    $this->filecache   = array_key_exists('file-cache', $o);
+    $this->pcreCache  = (string)hphp_array_idx($o, 'pcre-cache', 'static');
+    $this->pcreSize   = (int)hphp_array_idx($o, 'pcre-cache-size', 98304);
+    $this->pcreExpire = (int)hphp_array_idx($o, 'pcre-cache-expire', 7200);
+    $this->allVolatile   = array_key_exists('all-volatile', $o);
+    $this->interpPseudomains = array_key_exists('interp-pseudomains', $o);
+
+    if (array_key_exists('tcprint', $o)) {
+      $tcprint = hphp_array_idx($o, 'tcprint', null);
+      if (is_string($tcprint) && $tcprint !== '') {
+        $this->tcprint = $tcprint;
+      } else if ($isFacebook) {
+        $this->tcprint =
+          $fbcode . '/_bin/hphp/facebook/tools/tc-print/tc-print';
+      }
+    }
+    $this->tcAlltrans = array_key_exists('dump-all-trans', $o);
+    $this->tcToptrans = array_key_exists('dump-top-trans', $o);
+    $this->tcTopfuncs = array_key_exists('dump-top-funcs', $o);
+    $this->pcredump    = array_key_exists('dump-pcre-cache', $o);
```

Comment:
if you are going to use ascii art and line up the = on lines 215 and 216, also do that for lines 212 through 214, above